redhat-developer/try-in-dev-spaces-browser-extension
# GitHub Enterprise support

This PR adds GitHub Enterprise support by allowing users to specify additional GitHub domains to inject the button for.

Fixes #72 

<details>
<summary>New GitHub Enterprise domains tab</summary>

<img width="817" alt="image" src="https://github.com/redhat-developer/try-in-dev-spaces-browser-extension/assets/83611742/f14d37d5-85cc-44d6-ae2f-a8b0cb394d79">

</details>

To test this PR on Google Chrome:
1. Checkout the PR branch
2. Apply patch to prevent button injection on github.com:
```
curl https://gist.githubusercontent.com/dkwon17/77fed8fe67080c09866fe2c0b0ade350/raw/4bd128687bdd8dfb4c71c8c618a10eeebd3d7098/try-in-dev-spaces-patch | git apply
```
3. Build project:
```
yarn build
```
4. Sideload extension to Google Chrome (see [sideload instructions](https://github.com/redhat-developer/try-in-dev-spaces-browser-extension/blob/main/CONTRIBUTING.md))
5. Go to a GitHub project page and verify that the `Dev Spaces` button is not being injected
6. Go to the options page for the extension and add 'https://github.com' as a GitHub Enterprise domain. Next, go back to the GitHub repo page and refresh the page. The `Dev Spaces` button should be injected:

https://github.com/redhat-developer/try-in-dev-spaces-browser-extension/assets/83611742/fc9efc52-2422-481a-b52a-095328e1ca4c
